modules/swagger-parser-v3/src/main/java/io/swagger/parser/v3/OpenAPIV3Parser.java

```diff
@@ -27,12 +27,11 @@
     public SwaggerParseResult readLocation(String url, List<AuthorizationValue> auth, ParseOptions options) {
         SwaggerParseResult result = new SwaggerParseResult();
         try {
-            // TODO
             JsonNode node = YAML_MAPPER.readValue(new URL(url), JsonNode.class);
             if(node != null && node.get("openapi") != null) {
                 JsonNode version = node.get("openapi");
                 if(version.asText() != null && version.asText().startsWith("3.0")) {
-                    result = new OpenAPIDeserializer().deserialize(node);
+                    result.setOpenAPI(new OpenAPIResolver(new OpenAPIDeserializer().deserialize(node).getOpenAPI(),null,null).resolve());
```

Comment:
You need to pass the `auth` option along when doing the resolving, also consider the `options` in the call